FunctionSet.py

```diff
@@ -0,0 +1,181 @@
+        b = [abs(x - y) for x, y in zip(self.data['High'].iloc[:-1], self.data['Close'].iloc[1:])]
+        c = [abs(x - y) for x, y in zip(self.data['Low'].iloc[:-1], self.data['Close'].iloc[1:])]
+        return [max(x, y, z) for x, y, z in zip(a, b, c)]
+
+    def KD(self, days, columns = None, weight = (2/3, 1/3)):
+        data = pd.DataFrame(self.data)
+        if columns != None:
+            data = data[columns]
+        mx, mn = [], []
+        for i in range(data.shape[0]):
+            mx.append(max(data.iloc[i,:]))
+            mn.append(min(data.iloc[i,:]))
+        rs_h = pd.Series(mx).rolling(days).max() 
+        rs_l = pd.Series(mn).rolling(days).min()
+        RSV = [(z-y)/(x-y) for x, y, z in zip(rs_h.iloc[days-1:], rs_l.iloc[days-1:], data['Close'].iloc[days-1:])]
+        del mx, mn, rs_h, rs_l
+        K_val, D_val = [weight[1]*RSV[0]], [weight[1]*weight[1]*RSV[0]]
+        for i in range(1, len(RSV)):
+            K_val.append(weight[0]*K_val[i-1] + weight[1]*RSV[i])
+            D_val.append(weight[0]*D_val[i-1] + weight[1]*K_val[i])
+        return K_val, D_val
+        
+    def ATR(self, days):
+        atr = self.Ma(target = 'Close', days = days)
+        return atr
+
+    def DI(self):
+        return [(x + y + 2 * z)/4 for x, y, z in zip(self.data['High'], self.data['Low'], self.data['Close'])]
+
+
+    def MACD(self, days, short = 12, long = 26):
+        if short >= long:
+            print('short should be less than long')
+            return None
+        di = self.DI()
+        ema_s, ema_l = [sum(di[:short])/short], [sum(di[:long])/long]
+        for i in range(len(di)):
+            if short + 1 + i <= len(di) - 1:
+                ema_s.append((ema_s[-1] * (short - 1) + di[short + 1 + i] * 2) / (short + 1))
+                if long + 1 + i <= len(di) - 1:
+                    ema_l.append((ema_l[-1] * (long - 1) + di[long + 1 + i] * 2) / (long + 1))
+        ema_s = ema_s[long - short : ]
+        dif = [x - y for x, y in zip(ema_s , ema_l)]
+        macd = [sum(dif[:days])/days]
+        for i in range(len(dif[days:])):
+            macd.append((macd[-1] * (days - 1) + dif[days + i] * 2) / (days + 1))
+        return macd
+
+    def BIAS(self, target, days):
+        ma = self.Ma(self.data, target, days)[days:]
+        bias = [(x - y) / y for x, y in zip(self.data[target][days:], ma)]
+        return bias
+
+    def RSI(self, days):
+        sub = [x - y for x, y in zip(self.data['Open'], self.data['Close'])]
+        up = sum([x for x in sub if x >= 0])
+        dn = sum([abs(x) for x in sub if x < 0])
+        return up/dn
+
+class Action:
+    def __init__(self, data, initial, account = [], coin_hold = [], coin_owed = [], status = []):
+        self.data = data
+        self.initial = initial
+        self.account = account
+        self.coin_hold = coin_hold
+        self.coin_owed = coin_owed
+        self.status = status
+    
+    def get_coin_hold(self):
```

Comment:
why add get_ method ?!
If it is not private attribute, it conventionally needs no corresponding get_ method~